packages/eslint-plugin/tests/rules/space-infix-ops.test.ts

```diff
@@ -1022,7 +1022,7 @@ ruleTester.run('space-infix-ops', rule, {
         type Test=|string|(((() => void)))|string;
       `,
       output: `
-        type Test = |string | (((() => void))) | string;
+        type Test = | string | (((() => void))) | string;
```

Comment:
This should be the real fix. See [this playground](https://typescript-eslint.io/play/#ts=5.3.3&fileType=.ts&code=C4TwDgpgBAKhDOwoF4A%2BiBOBLAdgc1QApjCBKFAPigDcB7LAE1OfWG3wG4g&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKWQBwEMBjRAWgEsA7AM3IA9SB7PVDSR66yAGnG0gACAFwCeeFEWjk8Q0inhUhAenzEyVWg2asoiaNEbRIfAL4gTQA&tsconfig=&tokens=false).

The test currently fails because it's being fixed to:
```TS
type Test =  | string | (((() => void))) | string;
```
(2 spaces after the `=`)
The reason is probably because there are 2 fixes that add spaces before and after the relevant characters (`=` & `|`), which then results in 2 spaces.

Not sure how to approach this. I have 2 options in mind:
1. Trimming the spaces after each fix (for sure a bad idea)
2. Re-running the rule tester on the same line after each fix (is that what happens in real-life? not sure if/how to do it though)


WDYT?